src/app/model.ts

```diff
@@ -1,36 +1,133 @@
+}
 
-    const zdMessage = additionalMessage + '\n' +
-            postMessage + '\n' +
-            mmSignature;
+export class TicketFromForm implements IticketFromFrom {
+    formValues: AppFormValues;
+    ticket: Tickets.CreateModel;
+    fieldValidationErrors: any
 
-    const ticket: Tickets.CreatePayload = {
-        ticket: {
-            subject: values.subject,
-            type: values.type,
-            priority: values.priority,
+    constructor(values: AppFormValues) {
+        this.formValues = values;
+        this.fieldValidationErrors = {};
+        this.ticket = {
             comment: {
-                body: zdMessage,
+                body: this.getPostMessage(),
             },
-        },
-    };
-    return ticket;
-};
+        };
+    }
+
+    getPostMessage(): string {
+        const mmSignature = '*message created from Mattermost message.*\n' + configStore.getSiteURL();
+
+        const additionalMessage = this.formValues[fieldNames.additionalMessage] || '';
+        const postMessage = this.formValues[fieldNames.postMessage] || '';
+
+        const zdMessage = additionalMessage + '\n' +
+                postMessage + '\n' +
+                mmSignature;
+
+        return zdMessage;
+    }
+
+    getTicket(): Tickets.CreatePayload {
+        return this.mapFormValuesToTicket(this.ticket);
+    }
+
+    // mapFormValuesToTicket returns a zendesk ticket with AppCall values mapped
+    // 1:1 to a zendesk ticket
+    mapFormValuesToTicket(ticket: Tickets.CreateModel): Tickets.CreatePayload {
+        // app form values that are not to be 1:1 mapped to zendesk fields
+        const omitFields = [fieldNames.additionalMessage, fieldNames.postMessage];
+
+        // iterate through each field and build the Zendesk ticket
+        Object.keys(this.formValues).forEach((fieldName) => {
+            switch (true) {
+            case omitFields.includes(fieldName):
+                return;
+
+            // if a form is not defined, selected, or checked in the modal, its
+            // value will be null. continue to next field
+            case !this.formValues[fieldName]:
+                return;
+
+            // field is a custom field
+            case RegExp(`${fieldNames.customFieldPrefix}*`).test(fieldName):
+                this.handleCustomField(fieldName);
+                return;
 
-export default getTicketFromForm;
+            default:
+                // app form field names were mapped to a corresponding Zendesk field name. Save them
+                // directly to the ticket payload
+                ticket[fieldName] = this.getFieldValue(fieldName);
+            }
+        });
+
+        return {ticket};
+    }
+
+    handleCustomField(fieldName: string): void{
+        const getOption = (option) => (option.value);
+        const getMultiselectValues = (options) => options.map(getOption);
+
+        const re = new RegExp(fieldNames.customFieldPrefix);
+        if (re.test(fieldName)) {
```

Comment:
Already did the regents test to get into this function.  No need to test again.  Just deconstruct the values

Reply:
done.